Assignment4/Assignment4.js

```diff
@@ -0,0 +1,35 @@
+const fs = require('fs');
+
+function print(string) {
+  console.log(string);
+}
+
+function checkError(error) {
+  if (error) {
+    print(error);
+    throw error;
+  }
+}
+
+function readAppendWrite(readFile, appendFile, writeFile) {
+  fs.readFile(readFile, function (errorOnFile1Read, dataFromFile1) {
+    checkError(errorOnFile1Read); 
+    print(dataFromFile1.toString());
+    print("----------------------------------------------");
+
+    fs.appendFile(appendFile, dataFromFile1, function (errorOnFile2Append) {
+      checkError(errorOnFile2Append);
+  
+      fs.readFile(appendFile, function (errorOnFile2Read, dataFromFile2) {
+        checkError(errorOnFile2Read);
+        
+        fs.writeFile(writeFile, datafromFile2, function (errorOnFile3Write) {
```

Comment:
`datafromFile2` is not defined.

Read the detail of the error below.

-----
fs.writeFile(writeFile, datafromFile2, function (errorOnFile3Write) {

ReferenceError: datafromFile2 is not defined
    at /home/rails/work/training/aarushi_goyal/node-training/Assignment4/Assignment4.js:26:33
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:445:3)

-----